### ISSUE TITLE
diabetes

### ISSUE THREAD

**Issue by venky0116** [venky0116/EMC-project#1]

**Comment by venky0116** [comment]
import pandas as pd
import numpy as np
from sklearn.naive_bayes import GaussianNB
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
import json
# Load dataset (Pima diabetes)
df = pd.read_csv("diabetes.csv")
# Features and Target
X = df.drop("Outcome", axis=1)
y = df["Outcome"]

# Train-test split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.3, random_state=0
)

# Train Naive Bayes model
model = GaussianNB()
model.fit(X_train, y_train)

# Evaluate accuracy
y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
print("Model Accuracy:", accuracy)

# --------- Take user input for prediction ---------
print("\nEnter patient details to check diabetes risk (press Enter if unknown):")

def get_input(prompt, default_value):
    """Helper function: returns float(input) or default if blank"""
    val = input(prompt)
    if val.strip() == "":
        return default_value
    return float(val)

# Use dataset mean values if user skips
defaults = X.mean()

pregnancies = get_input("Number of Pregnancies: ", defaults["Pregnancies"])
glucose = get_input("Glucose Level: ", defaults["Glucose"])
blood_pressure = get_input("Blood Pressure: ", defaults["BloodPressure"])
skin_thickness = get_input("Skin Thickness: ", defaults["SkinThickness"])
insulin = get_input("Insulin Level: ", defaults["Insulin"])
bmi = get_input("BMI: ", defaults["BMI"])
dpf = get_input("Diabetes Pedigree Function: ", defaults["DiabetesPedigreeFunction"])
age = get_input("Age: ", defaults["Age"])

# Prepare input for prediction
new_data = np.array([[pregnancies, glucose, blood_pressure,
                      skin_thickness, insulin, bmi, dpf, age]])

# Predict
prediction = model.predict(new_data)

if prediction[0] == 1:
    print("✅ The person is likely to have Diabetes.")
else:
    print("❌ The person is NOT likely to have Diabetes.")
